### PR TITLE
Add FalkorDB  graph database

### PR DESCRIPTION
**Pull request by galshubeli**
- Added the FalkorDB graph stores

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,11 +53,15 @@ memary will default to the locally run models unless explicitly specified. Addit
...
    ```
    OPENAI_API_KEY="YOUR_API_KEY"
-   NEO4J_PW="YOUR_NEO4J_PW"
-   NEO4J_URL="YOUR_NEO4J_URL"
    PERPLEXITY_API_KEY="YOUR_API_KEY"
    GOOGLEMAPS_API_KEY="YOUR_API_KEY"
    ALPHA_VANTAGE_API_KEY="YOUR_API_KEY"
+   
+   Database usage:
+   FALKORDB_URL="YOUR_FALKORDB_URL"
```
Comment (swilly22) [inline README.md:61]: Provide an example format. e.g. <username:password@host:ip>

File: README.md [hunk 2]
```diff
@@ -68,6 +72,7 @@ memary will default to the locally run models unless explicitly specified. Addit
      <summary>API Info</summary>
 
     - [**OpenAI key**](https://openai.com/index/openai-api)
+    - [**FalkorDB**](https://www.falkordb.com/try-free/)
```
Comment (swilly22) [inline README.md:75]: https://app.falkordb.cloud/signin ?

File: src/memary/agent/base_agent.py [hunk 3]
```diff
@@ -85,14 +78,27 @@ def __init__(
...
-            password=self.neo4j_password,
-            url=self.neo4j_url,
-            database=database,
-        )
+        
+        self.falkordb_url = os.getenv("FALKORDB_URL")
+        if self.falkordb_url is not None:
+            from llama_index.graph_stores.falkordb import FalkorDBGraphStore
+            # initialize FalkorDB graph resources
+            self.graph_store = FalkorDBGraphStore(self.falkordb_url,  decode_responses=True)
```
Comment (swilly22) [inline src/memary/agent/base_agent.py:86]: remove extra space between `self.falkordb_url` and `decode_responses=True`

File: streamlit_app/app.py [hunk 4]
```diff
@@ -65,22 +67,38 @@ def create_graph(nodes, edges):
...
+    
+    if chat_agent.falkordb_url is not None:
+        falkordb = FalkorDB.from_url(chat_agent.falkordb_url)
+        session = falkordb.select_graph('falkor')
+        result = session.query(cypher_query).result_set
+        for record in result:
+            path = record[0]
+            n1_id = re.search(r'\{id:(.*?)\}', path.get_node(0).to_string()).group(1)
+            n2_id = re.search(r'\{id:(.*?)\}', path.get_node(1).to_string()).group(1)
+            rels = [re.search(r':(.*?)\]', str(path.get_edge(i))).group(1) for i in range(path.edge_count())]
```
Comment (swilly22) [inline streamlit_app/app.py:79]: Can you please check and share here what's the value of `cypher_query`
this is probably not the way to extract items from the result-set.

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: src/memary/agent/chat_agent.py [hunk 5]
```diff
@@ -18,7 +18,7 @@ def __init__(
         user_persona_txt,
         past_chat_json,
         llm_model_name="llama3",
-        vision_model_name="llava",
+        vision_model_name="gpt-4-vision-preview",
```
[the remark this answers is not in the document]
Comment (galshubeli) [reply src/memary/agent/chat_agent.py:21]: No good reason, I switched it back.

File: streamlit_app/app.py [hunk 6]
```diff
@@ -65,22 +66,38 @@ def create_graph(nodes, edges):
...
-                rels = [rel.type for rel in path.relationships]
-
-                n1_id = record["p"].nodes[0]["id"]
-                n2_id = record["p"].nodes[1]["id"]
-                nodes.add(n1_id)
-                nodes.add(n2_id)
-                edges.append((n1_id, n2_id, rels))
-                entities.extend([n1_id, n2_id])
+    
+    if chat_agent.falkordb_url is not None:
```
[the remark this answers is not in the document]
Comment (galshubeli) [reply streamlit_app/app.py:70]: Agree

**Review by kevinl424 — APPROVED** [review]

**Pull request merged**